autogen.sh

```diff
@@ -23,6 +23,13 @@
 
 PROJECT="libva"
 
+# for `meson dist`
+if test -z "$srcdir"; then
+    srcdir="$MESON_PROJECT_DIST_ROOT"
+    test -n "$srcdir" || srcdir="$MESON_DIST_ROOT"
+    NOCONFIGURE=1
```

Comment:
As caught by ci, this line breaks typically autotools build flow many people have:
```
$ ./autogen.sh
<...>
$ make
make: *** No targets specified and no makefile found.  Stop.
```

Can we test if we under meson upfront checking existence of some meson specific environment variable and avoid breaking autotools experience? Maybe:
```
if test -n "$MESON_PROJECT_DIST_ROOT"
<...>
fi
```

Reply:
We can check if the previous srcdir= assignment succeeded, I guess. ;)

Reply:
I will add that the autotools flow many people have is broken and wrong and utterly incorrect -- people should never ever do this under any circumstances.

This is of course a personal opinion and I didn't intend to break it.

Reply:
> We can check if the previous srcdir= assignment succeeded, I guess. ;)

Works for me.